### ISSUE TITLE
Fails to map parameter

### ISSUE THREAD

**Issue by salsolatragus** [stg-tud/MUDetect#8]
We fail to map the parameter in http://mubench.svamann.de/fse18-mudetect-dev/experiments/2/detectors/MuDetect_180227_1_ExtendOnlyDataFlow/projects/itext/versions/5091/misuses/finding-8

**Comment by nguyenhoan** [comment]
Is this because the parameter type is always PdfArray in all instances of the pattern but PdfStructureElement  in this target?

**Comment by salsolatragus** [comment]
We're using the same label for all data nodes in the matching, so this shouldn't be the problem, as the difference only shows in the display output...

**Comment by nguyenhoan** [comment]
I remember that in detection, you use literal values when checking data node matches.
I thought you also you types when matching non-literal data nodes in similar way.

**Comment by salsolatragus** [comment]
No, we only ever use the abstracted data-node labels. All data nodes are objects. Everything else introduced too much noise. This didn't change since ICSE. We talked about using concrete names for  constants at some point, but I never got around to actually test that and I won't, at this point ;)
